hazelcast/connection.py

```diff
@@ -438,10 +438,23 @@ def _get_or_connect(self, address):
                         error = sys.exc_info()
                         return ImmediateExceptionFuture(error[1], error[2])
 
-                    future = self._authenticate(connection).continue_with(
-                        self._on_auth, connection, address
-                    )
+                    future = Future()
                     self._pending_connections[address] = future
+
+                    def cb(response):
+                        self._on_auth(response, connection, address, future)
```

Comment:
why do you pass future here as last parameter? I checked `continue_with()`,  a future not passed there as last parameter. Am I wrong?